File: src/main/java/br/org/acal/resouces/model/PartnerModel.java
package br.org.acal.resouces.model;

import jakarta.persistence.Column;
import jakarta.persistence.Entity;
import jakarta.persistence.FetchType;
import jakarta.persistence.Id;
import jakarta.persistence.JoinColumn;
import jakarta.persistence.OneToOne;
import jakarta.persistence.Table;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.io.Serializable;

@Entity
@Table(name = "socio")
@Data
@AllArgsConstructor
@NoArgsConstructor
@Builder
public class PartnerModel implements Serializable {

    @Id
    @Column(name = "id")
    private String number;

    @Column(name = "SocioExclusivo")
    private Boolean memberOnly;

    @Column(name = "numeroSocio")
    private int partnerNumber;

    @OneToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "idPessoa", referencedColumnName = "id")
    private CustomerModel customer;

}

File: src/main/kotlin/br/org/acal/resources/model/PartnerModel.kt
package br.org.acal.resources.model

import jakarta.persistence.Column
import jakarta.persistence.Entity
import jakarta.persistence.Id
import jakarta.persistence.JoinColumn
import jakarta.persistence.OneToOne
import jakarta.persistence.Table
import java.io.Serializable
import lombok.AllArgsConstructor
import lombok.Builder
import lombok.Data
import lombok.NoArgsConstructor

@Entity
@Table(name = "socio")
@Data
@AllArgsConstructor
@NoArgsConstructor
@Builder
class PartnerModel(

    @Id
    @Column(name = "id")
    val number: String,

    @Column(name = "SocioExclusivo")
    val memberOnly: Boolean,

    @Column(name = "numeroSocio")
    val partnerNumber: Number,

    @OneToOne
    @JoinColumn(name = "idPessoa", referencedColumnName = "id")
    val customer: CustomerModel,

) : Serializable
